File: system-tests/multiple-agents-test/src/test/java/com/teamscale/client/MultipleAgentsTest.java
package com.teamscale.client;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class MultipleAgentsTest {

	private static final Path LOG_DIRECTORY = Paths.get("logTest").resolve("logs");

	@Test
	public void systemTest() throws IOException {
		assertThat(LOG_DIRECTORY).exists();
		List<String> lines = Files.readAllLines(LOG_DIRECTORY.resolve("teamscale-jacoco-agent.log"));
		assertThat(lines).anyMatch(
				s -> s.contains("Using multiple java agents could interfere with coverage recording.")
		);
		assertThat(lines).anyMatch(
				s -> s.contains("For best results consider registering the Teamscale JaCoCo Agent first.")
		);
	}
}

File: system-tests/multiple-agents-test/src/test/kotlin/com/teamscale/client/MultipleAgentsTest.kt
package com.teamscale.client

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.io.IOException
import java.nio.file.Paths
import kotlin.io.path.readLines

class MultipleAgentsTest {
	@Test
	@Throws(IOException::class)
	fun systemTest() {
		assertThat(LOG_DIRECTORY).exists()
		val lines = LOG_DIRECTORY.resolve("teamscale-jacoco-agent.log").readLines()
		assertThat(lines)
			.anyMatch { it.contains("Using multiple java agents could interfere with coverage recording.") }
		assertThat(lines)
			.anyMatch { it.contains("For best results consider registering the Teamscale JaCoCo Agent first.") }
	}

	companion object {
		private val LOG_DIRECTORY = Paths.get("logTest").resolve("logs")
	}
}
